### PR TITLE
[KOGITO-2374] Setup Maven repo url

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2374

Consider the given repo-url in local and remote builds.

- Refactored a bit the way scripts are updating version / env
- `KOGITO_MAVEN_REPO_URL` is added if needed to change the default Kogito artifacts url (`https://repository.jboss.org/nexus/content/groups/public/`).
- `clone-repos.sh` is using the `configure` & `settings.xml` from kogito-maven

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by sutaakar** [comment]
I am not sure that we should use `MAVEN_REPO_URL` for our JBoss repo.
What will happen in case user will have some domain artifacts built in his own repository and will want to use both JBoss (for common Kogito artifacts) and his own repository in the build?

**Comment by radtriste** [comment]
Good point but the user should use normally the Maven central artifacts by default and not the JBoss ones or ?
Jboss one should be used only for snapshot then.

Another concern is that if we have a staging repository or another repository containing the latest snapshot artifacts and JBoss repo is not yet updated (a mirror repository takes some time for the update...), who has precedence on the other ?
Will I get the latest snapshots ?
Or we should have a way to add and override a KOGITO_MAVEN_URL env ?
and leave the MAVEN_REPO_URL free ?

Another way would be to leverage the `MAVEN_REPOS` extension

**Comment by spolti** [comment]
I am not sure if remove the default repo is a good idea, maybe use -U to force download the latests artifacts?

**Comment by radtriste** [comment]
@spolti yes and no because if I have 2 repositories both with those snapshots, the first one tried and answering ok will give the snapshots, and may be not up to date (mirroring in Maven can take time...)
But maybe I can define a KOGITO_MAVEN_URL with JBoss default which could be overriden ?

btw, for productive images, shouldn't the default repository be Maven central ?

**Comment by spolti** [comment]
maybe disable the snapshots on jboss central might be better.

> btw, for productive images, shouldn't the default repository be Maven central ?

not really, jboss repo has been in use since beginning cloud effort, if there's a strong reason to change that, we can.

> But maybe I can define a KOGITO_MAVEN_URL with JBoss default which could be overriden ?

Even better, we can remove the jboss central from settings.xml and, during build time, if no repo is set at all, we define the default one. this way, no other repo will be added if you correctly set the maven configuration envs

**Comment by radtriste** [comment]
so will go for being able to change default Maven repo via `KOGITO_MAVEN_REPO_URL` and `MAVEN_REPO_URL` keeps its old behavior.

**Comment by radtriste** [comment]
@spolti @sutaakar update to `KOGITO_MAVEN_REPO_URL` done. let me know what you think

File: modules/kogito-maven/3.6.x/module.yaml [hunk 1]
```diff
@@ -29,8 +29,11 @@ envs:
   - name: "MAVEN_DOWNLOAD_OUTPUT"
     description: "If set to true will print the transfer logs for downloading/uploading of maven dependencies. Defaults to false"
     example: "true"
+  - name: "KOGITO_MAVEN_REPO_URL"
```
Comment (sutaakar) [inline modules/kogito-maven/3.6.x/module.yaml]: Wouldn't it have more sense to name it something like JBOSS_MAVEN_REPO_URL ?
As it contains also other artifacts than just Kogito ones.

**Review by sutaakar — APPROVED** [review]
Looks good, one suggestion above.

File: modules/kogito-maven/3.6.x/module.yaml (hunk 1 above)
Comment (radtriste) [reply modules/kogito-maven/3.6.x/module.yaml]: will rename it

**Comment by radtriste** [comment]
@spolti @Kaitou786 can you please review ?

**Comment by Kaitou786** [comment]
@radtriste shouldn't we also change the  `update-artifacts.py` → `update-maven-information.py` in `Jenkinsfile-Nightly` I know it is deprecated but we are also keeping it in place as the whole nightly is a bit unstable

**Comment by radtriste** [comment]
you are right. Missed it. I will update it.

**Comment by radtriste** [comment]
@Kaitou786 updated, as well as the README

**Comment by Kaitou786** [comment]
Thanks :)

**Review by Kaitou786 — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Pull request merged**